File: src/main/java/com/zl/mjga/platform/events/EventPublicationOperations.java
package com.zl.mjga.platform.events;

import java.time.Duration;
import java.util.List;
import java.util.UUID;

/** Protected application interface for inspecting and recovering durable event publications. */
public interface EventPublicationOperations {

  List<PublicationSummary> incomplete(int limit);

  RecoveryOutcome resubmit(Duration minimumAge, int limit, InvocationContext context);

  RecoveryOutcome skip(UUID publicationId, String reason, InvocationContext context);

  record PublicationSummary(
      UUID publicationId, String listenerId, String eventType, int attempts, String status) {}

  record RecoveryOutcome(int affected, String outcome) {}
}

File: apps/server/src/main/kotlin/com/zl/mjga/platform/events/EventPublicationOperations.kt
package com.zl.mjga.platform.events

import java.time.Duration
import java.util.UUID

/** Protected application interface for inspecting and recovering durable event publications. */
interface EventPublicationOperations {
    fun incomplete(limit: Int): List<PublicationSummary>

    fun resubmit(
        minimumAge: Duration,
        limit: Int,
        context: InvocationContext,
    ): RecoveryOutcome

    fun skip(
        publicationId: UUID,
        reason: String,
        context: InvocationContext,
    ): RecoveryOutcome

    data class PublicationSummary(
        val publicationId: UUID,
        val listenerId: String,
        val eventType: String,
        val attempts: Int,
        val status: String,
    )

    data class RecoveryOutcome(
        val affected: Int,
        val outcome: String,
    )
}
